nlb/models/blockworld/blockworld.py

```diff
@@ -0,0 +1,180 @@
+import os
+import pathlib
+from typing import Callable
+
+import numpy as np
+import rich_click as click
+from openai.types import shared
+
+from nlb.models.blockworld import closed_loop_llm
+from nlb.models.blockworld import environment
+from nlb.models.blockworld import heuristic
+from nlb.models.blockworld import heuristic_llm
+from nlb.models.blockworld import mdp_llm
+from nlb.models.blockworld import mdp_manual
+from nlb.models.blockworld import metrics
+from nlb.models.blockworld import open_loop_llm
+from nlb.models.blockworld import plot
+from nlb.util import click_utils
+from nlb.util import console_utils
+from nlb.util import dataframe
+from nlb.util import path_utils
+from nlb.util import secrets
+from nlb.util import timer
+
+
+def _save_results(
+    results: list[metrics.Result], output: pathlib.Path, console: console_utils.Console
+) -> None:
+    df = dataframe.dataframe_from_type(metrics.Result, results)
+    df.to_csv(output, index=False)
+    console.info(f'Results saved to {output.relative_to(path_utils.REPO_ROOT)}.')
+
+
+@click.command()
+@click.option(
+    '--num-blocks',
+    '-n',
+    type=int,
+    default=5,
+    help='Number of blocks in the environment',
+)
+@click.option(
+    '--monte-carlo',
+    '-m',
+    type=int,
+    help='Number of monte carlo runs to average over (default: 1)',
+)
+@click.option(
+    '--policy',
+    '-p',
+    type=click_utils.EnumChoice(metrics.Policy),
+    multiple=True,
+    help='Policy(ies) to use',
+)
+@click.option(
+    '--make-gif',
+    is_flag=True,
+    help='Whether to create a GIF of the environment execution (1 seed per sim)',
+)
+def main(
+    num_blocks: int,
+    monte_carlo: int | None,
+    policy: list[metrics.Policy],
+    make_gif: bool,
+) -> None:
+    cur_dir = pathlib.Path(__file__).parent
+    console = console_utils.Console()
+
+    results: list[metrics.Result] = []
+
+    # Setup the OpenAI environment
+    os.environ['OPENAI_API_KEY'] = secrets.get_secret('openai.key')
+    model = 'gpt-5-mini'
+    reasoning_efforts: list[shared.ReasoningEffort] = [
+        'minimal',
+        'low',
+        'medium',
+        'high',
+    ]
+
+    try:
```

Comment:
this got pretty sloppy but we're hitting a diminishing return on what is fundamentally a homework assignment